adapters/socketio_berkeley.c

```diff
@@ -910,32 +910,18 @@ int socketio_send(CONCRETE_IO_HANDLE socket_io, const void* buffer, size_t size,
                 ssize_t send_result = send(socket_io_instance->socket, buffer, size, 0);
                 if ((send_result < 0) || ((size_t)send_result != size))
                 {
-                    if (send_result == INVALID_SOCKET)
+                    if (errno != EAGAIN)
                     {
-                        if (errno == EAGAIN) /*send says "come back later" with EAGAIN - likely the socket buffer cannot accept more data*/
-                        {
-                            /* queue data */
-                            if (add_pending_io(socket_io_instance, buffer, size, on_send_complete, callback_context) != 0)
-                            {
-                                LogError("Failure: add_pending_io failed.");
-                                result = MU_FAILURE;
-                            }
-                            else
-                            {
-                                /*do nothing*/
-                                result = 0;
-                            }
-                        }
-                        else
-                        {
-                            LogError("Failure: sending socket failed. errno=%d (%s).", errno, strerror(errno));
-                            result = MU_FAILURE;
-                        }
+                        LogError("Failure: sending socket failed. errno=%d (%s).", errno, strerror(errno));
```

Comment:
>. errno=%d (%s).", errno, strerror(errno)); [](start = 64, length = 43)

Do we care how many bytes it sent?  I'm not sure.

Reply:
I think according to the documentation, an error => no bytes sent.

---
In reply to: [326645677](https://github.com/Azure/azure-c-shared-utility/pull/380#discussion_r326645677) [](ancestors = 326645677)

Reply:
Do we also need to handle EWOULDBLOCK because socket is created in with non-blocking flag?

---
In reply to: [326714516](https://github.com/Azure/azure-c-shared-utility/pull/380#discussion_r326714516) [](ancestors = 326714516,326645677)

Reply:
I agree we should perhaps add EWOULDBLOCK, but that is an additional change in a hot path, and EAGAIN has sufficed for all the Linux platforms we use. After offline discussion, we agreed on not changing that how, since would require extra validation across all linuxes we support.

Reply:
@rafilho , it has been confirmed that on error no bytes are sent.